app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt

```diff
@@ -506,6 +449,96 @@ class AdministratorControlsFragmentTest {
     }
   }
 
+  private fun clickAutoUpdateTopicContainer() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.auto_update_topic_constraint_layout
+      )
+    ).perform(click())
+  }
+
+  private fun checkUpdateOnWifiSwitchNotClickable() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.topic_update_on_wifi_switch
+      )
+    ).check(matches(not(isClickable())))
+  }
+
+  private fun checkAutoUpdateTopicSwitchNotClickable() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.auto_update_topic_switch
+      )
+    ).check(matches(not(isClickable())))
+  }
+
+  private fun checkUpdateOnWifiSwitchNotChecked() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.topic_update_on_wifi_switch
+      )
+    ).check(matches(not(isChecked())))
+  }
+
+  private fun clickAutoUpdateTopicSwitch() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.auto_update_topic_switch
+      )
+    ).perform(click())
+  }
+
+  private fun checkAutoUpdateTopicSwitchIsChecked() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.auto_update_topic_switch
+      )
+    ).check(matches(isChecked()))
+  }
+
+  private fun checkAutoUpdateSwitchIsUnchecked() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.auto_update_topic_switch
+      )
+    ).check(matches(not(isChecked())))
+  }
+
+  private fun checkUpdateOnWifiSwitchIsChecked() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.topic_update_on_wifi_switch
+      )
+    ).check(matches(isChecked()))
+  }
+
+  private fun clickUpdateOnWifiSwitch() {
+    onView(
+      atPositionOnView(
+        recyclerViewId = R.id.administrator_controls_list,
+        position = 2,
+        targetViewId = R.id.topic_update_on_wifi_constraint_layout
+      )
```

Comment:
You could probably move this bit to a helper, too, to simplify all of these helper methods (I think they'd fit on just one line at that point).

Reply:
I think they cannot be subdivided further as they are in a single view assertion and perform state?

Reply:
You could do something like:

```kotlin
private fun clickUpdateOnWifiSwitch() {
  onView(atAdminControlsItem(position = 2, R.id.topic_update_on_wifi_constraint_layout)).perform(click())
}

private fun atAdminControlsItem(position: Int, @ResId viewId: Int): Matcher<View> {
  return atPositionOnView(recyclerViewId = R.id.administrator_controls_list, position, viewId)
}
```

(As an example). It doesn't quite fit on one line for the first bit, but there are other ways you could simplify things if you'd want to. FWIW I think this is optional, but it can help with test readability.

Reply:
done